### ISSUE TITLE
Error with truehd

### ISSUE THREAD

**Issue by arontx** [mdhiggins/sickbeard_mp4_automator#794]
When I add truehd to my list of audio codecs, I get the following error:

    Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument

Here is the full command:

    /usr/bin/ffmpeg -i /path/to/sample.mkv -vcodec libx264 -map 0:0 -c:a:0 aac -map 0:1 -disposition:a:0 default -ac:a:0 2 -b:a:0 256k -metadata:s:a:0 language=eng -strict experimental -c:a:1 copy -map 0:1 -metadata:s:a:1 language=eng -disposition:a:1 none -f mp4 -threads auto -y /path/to/sample.m4v

And here is what the relevant section of autoProcess.ini looks like:

    audio-codec = ac3,aac,eac3,dts,truehd

**Comment by Collisionc** [comment]
.mp4 containers do not support truehd audio streams, unfortunately. I'm in the process of making a pull request to 'fix' this by either forcing encoding of truehd to aac or spitting out a warning for people. I'll probably just add an option.

**Comment by Collisionc** [comment]
Opened pull request to resolve this - https://github.com/mdhiggins/sickbeard_mp4_automator/pull/816

Skipping the truehd stream seems to be the best option, as the next stream is always the AC3 core.  I've tried converting truehd --> aac/ac3/etc and the end result has issues with video/audio sync seemingly 50% of the time, so it's not reliable.

**Comment by arontx** [comment]
This is great, thanks Alex!

**Comment by sammys** [comment]
I'm a little late to pipe in but I figured I'd contribute to the discussion a little bit. I use sickbeard_mp4_automator to prepare things for Plex and no longer use iTunes so I have less restrictions. Until TrueHD came along I was able to keep original audio intact (a requirement I choose to have). In order to keep TrueHD I would need to expand sickbeard_mp4_automator to output an mkv instead of mp4 when a TrueHD stream is encountered.
